docs/blog/posts/2017-03-01-hermetic-builds.md

```diff
@@ -1,10 +1,19 @@
 ---
-title:  "Hermetic Builds with Robolectric"
-categories: build
-author: Christian Williams
+date: 2017-03-01
+authors:
+  - xian
+hide:
+  - footer
+  - toc
+slug: hermetic-builds
 ---
+
+# Hermetic Builds with Robolectric
+
 Robolectric needs access to multiple Android SDK jars in order to perform its magic, which means it needs special configuration beyond just setting up dependencies in your build. By default, it tries to download Android SDK jars from Maven Central.
 
+<!-- more -->
```

Comment:
This tells the blog plugin where to truncate the post on list of blog posts. On the screenshot below, `<!-- more -->` is replaced by `Continue reading`:
<img width="970" alt="Screenshot 2024-04-26 at 08 28 47" src="https://github.com/robolectric/robolectric.github.io/assets/1009664/57dd1529-bb1f-4f39-b761-dfc5b99c633b">

The default behaviour is to display the whole blog post.
I've made this tag mandatory, to avoid mistakes:
https://github.com/MGaetan89/robolectric.github.io/blob/7d24936f7d67ac0382315bfdfeee59587ae6a51c/mkdocs.yml#L58

I can make it optional if you're fine having whole post on the main page. Or the tag can be customised, if you prefer to use something else.

Reply:
You mean automatically detect where to cut the post?

Not from what I could find in the documentation:
https://squidfunk.github.io/mkdocs-material/plugins/blog/#config.post_excerpt
> When a post doesn't define an excerpt, views include the entire post

Reply:
I can try to override this template, to create this behaviour on our side:
https://github.com/squidfunk/mkdocs-material/blob/master/src/templates/partials/post.html

Reply:
Done!

It's no longer necessary to add `<!-- more -->` in each blog post.
Instead, I've overriding the `partials/post.html` template to display only the first paragraph:
https://github.com/robolectric/robolectric.github.io/blob/62404bbf02b29018b5ae554b7eaf285ac075340b/overrides/partials/post.html#L90-L94